### DISCUSSION TITLE
Test access to private repository

### DISCUSSION THREAD

**Discussion by bruchar1** [python-poetry/poetry#9200] [Q&A]
I'd like to be able to test whether the use has access to a given private repository. The context is that I call poetry from a script, and I want to display a proper explanation if the used did not configured an access token for the private repository.

My first idea was to check if `pypi-token.myrepo` config was set, but those config keys are hidden.
My second idea was to to call `poetry search privatepackage` but I didn't find a way to search on a specific repository.
So, am I missing something obvious, or is it because it is not possible actually?

**Comment by mogsdad** [comment]
We've got something that works for testing whether the dev is currently authorized for our private repository, relying on `poetry add`.

``` bash
poetry add --no-cache --dry-run wheel >/dev/null 2>&1 || echo "Authz for poetry repository expired" && exit 1
```

This fails quickly when authorization has expired, but the length of time it takes when authz is current depends on the the number of packages in your dependencies -- far too long. It would be better if there was a dedicated authz check built in.
